### PR TITLE
Make Glint apply to safe-tools-client hbs templates

### PR DESCRIPTION
**Pull request by lukemelia**

### CODE REVIEW

**Review by jurgenwerk — APPROVED** [review]

File: packages/safe-tools-client/app/components/safe-info/index.gts [hunk 1]
```diff
@@ -15,26 +15,29 @@ interface Signature {
...
   Args: {
     currentSafe: Safe | undefined;
-    safes: Safe[];
-    tokenBalances: TokenBalance[];
+    safes: Safe[] | undefined;
+    tokenBalances: TokenBalance[] | undefined;
     safesLoadingError: Error | undefined;
     onDepositClick: () => void;
-    onSelectSafe: () => void;
+    onSelectSafe: (safe: any) => void;
```
Comment (danibonilha) [inline packages/safe-tools-client/app/components/safe-info/index.gts:22]: Isn't (safe:Safe)?

**Review by danibonilha — APPROVED** [review]
Nice! Just some small nits

File: packages/safe-tools-client/app/components/deposit-modal/index.gts [hunk 2]
```diff
@@ -28,7 +28,7 @@ interface Signature {
   Args: {
     isOpen: boolean;
     onClose: () => void;
-    safeAddress: string;
+    safeAddress: string | undefined;
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/safe-tools-client/app/components/deposit-modal/index.gts:31]: I think there is a subtle difference. `parameter?: type` is the equivalent to `parameter: type | undefined = undefined`. We want glint to require you to specify the `safeAddress` argument when using this component, but we want to be flexible enough to accept an undefined value being passed.

File: packages/safe-tools-client/app/components/safe-info/index.gts [hunk 3]
```diff
@@ -15,26 +15,29 @@ interface Signature {
   Element: HTMLDivElement;
   Args: {
     currentSafe: Safe | undefined;
-    safes: Safe[];
-    tokenBalances: TokenBalance[];
+    safes: Safe[] | undefined;
+    tokenBalances: TokenBalance[] | undefined;
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/safe-tools-client/app/components/safe-info/index.gts:19]: 👍

File: packages/safe-tools-client/app/components/safe-info/index.gts (hunk 1 above)
Comment (lukemelia) [reply packages/safe-tools-client/app/components/safe-info/index.gts]: 👍

File: packages/safe-tools-client/app/components/safe-info/index.gts (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/safe-tools-client/app/components/safe-info/index.gts:19]: Ditto response, too :)

**Pull request merged**